crates/shielded_token/src/masp/shielded_sync/dispatcher.rs

```diff
@@ -381,7 +382,101 @@ where
         }
     }
 
-    fn apply_cache_to_shielded_context(
+    // Search for the Transaction order usedby the protocol using trial and
+    // error.
+    async fn transaction_order_search(
+        &mut self,
+        last_witnessed_tx: &Option<IndexedTx>,
+    ) -> Result<Vec<(IndexedTx, Transaction)>, eyre::Error> {
+        let mut ordered_txs = vec![];
+        // Get the unordered Transactions with their indices
+        let mut indexed_txs = self.cache.fetched.take().into_iter().peekable();
+        let needs_witness_map_update =
+            self.client.capabilities().needs_witness_map_update();
+        while let Some(first_tx) = indexed_txs.next() {
+            // Take only Transactions belonging to one block
+            let mut block_txs = vec![first_tx];
+            while let Some(next_tx) = indexed_txs
+                .next_if(|next_tx| next_tx.0.height == block_txs[0].0.height)
+            {
+                block_txs.push(next_tx);
+            }
+            // Do not apply these Transactions if they have been applied before
+            if !needs_witness_map_update
```

Comment:
This method will return an empty vec of `ordered_txs` if `needs_witness_map_update = true`, since we are skipping all updates of that vec from the outer-most loop (we hit the `continue` branch). This means that the indexer client is broken with this patch.

I think we should skip calling `transaction_order_search` from the indexer client, anyway, because the commitment tree isn't built locally, but rather fetched from the indexer. That also means we don't need to modify the `IndexerMaspClient` with an additional tendermint client field, because in practice it will never be needed.

Reply:
After employing a similar hack to the masp indexer, the `/tx` endpoint should be returning the correct execution order, so I think we can restrict the ordering shenanigens to the ledger client (i.e. the client that solely utilizes the cometbft rpc).